src/dbms/replication_handler.cpp

```diff
@@ -259,4 +265,177 @@ void RestoreReplication(replication::ReplicationState &repl_state, storage::Stor
       },
       repl_state.ReplicationData());
 }
+
+namespace system_replication {
+#ifdef MG_ENTERPRISE
+void SystemHeartbeatHandler(const uint64_t ts, slk::Reader *req_reader, slk::Builder *res_builder) {
+  replication::SystemHeartbeatReq req;
+  replication::SystemHeartbeatReq::Load(&req, req_reader);
+  memgraph::slk::Load(&req, req_reader);
```

Comment:
Do you need both these?

Reply:
Shouldn't SystemHeartbeatReq call slk in case it needs to call it?

Reply:
Good catch!!